app/src/main/kotlin/com/wire/android/util/ImageUtil.kt

```diff
@@ -45,20 +45,31 @@ object ImageUtil {
         Medium, Small
     }
 
-
     /**
      * Attempts to read the width and height of an image represented by the input parameter
      */
-    fun extractImageWidthAndHeight(imageDataInputStream: InputStream, mimeType: String): Pair<Int, Int> {
+    fun extractImageWidthAndHeight(path: String, mimeType: String): Pair<Int, Int> {
         val isAnimated = mimeType.contains("gif") || mimeType.contains("webp")
         if (isAnimated) {
-            BitmapFactory.decodeStream(imageDataInputStream).let { bitmap ->
+            BitmapFactory.decodeStream(File(path).inputStream()).let { bitmap ->
                 return bitmap.width to bitmap.height
             }
         } else {
-            val exifInterface = ExifInterface(imageDataInputStream)
-            val exifWidth: Int = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_WIDTH, 0)
-            val exifHeight: Int = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_LENGTH, 0)
+            val exifInterface = ExifInterface(File(path))
+            // we had a fallback to get the width and height from the file because the ExifInterface is not returning
+            // values after edit some images
+            val exifWidth: Int = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_WIDTH, 0).let {
+                if (it != 0) {
+                    return@let it
+                }
+                BitmapFactory.decodeFile(path).width
+            }
+            val exifHeight: Int = exifInterface.getAttributeInt(ExifInterface.TAG_IMAGE_LENGTH, 0).let {
+                if (it != 0) {
+                    return@let it
+                }
+                BitmapFactory.decodeFile(path).height
+            }
```

Comment:
Maybe we could simplify this method with the following:
```suggestion
fun extractImageWidthAndHeight(kaliumFileSystem: KaliumFileSystem, imageDataPath: Path): Pair<Int, Int> {
        kaliumFileSystem.source(imageDataPath).buffer().use { bufferedSource ->
            BitmapFactory.decodeStream(bufferedSource.inputStream()).let { bitmap ->
                return bitmap.width to bitmap.height
            }
        }
    }
```